librz/core/cprint.c

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2022 Peiwei Hu <jlu.hpw@foxmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+
+#define STRING_CHUNK 16
+
+/**
+ * Return a C/C++ string defination with block size as the length
+ * \param core RzCore
+ * \return a string defination or NULL if the error happens
+ */
+RZ_API char *rz_core_print_string_c_cpp(RzCore *core) {
+	ut64 value;
+	size_t size = core->blocksize;
+	RzStrBuf *sb = rz_strbuf_new(NULL);
+
+	if (!sb) {
+		RZ_LOG_ERROR("Fail to allocate the memory\n");
+		return NULL;
+	}
+	rz_strbuf_appendf(sb, "#define STRING_SIZE %" PFMTSZd "\nconst char s[STRING_SIZE] = \"", size);
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos && !(pos % STRING_CHUNK)) {
+			// newline and padding for long string
+			rz_strbuf_appendf(sb, "\"\n                            \"");
```

Comment:
Why the spaces after the newline?

Reply:
They are for pretty print. These spaces correspond to `const char s[STRING_SIZE] = ` in the first line. Otherwise, the new line will start from the line head.

Reply:
```
#define STRING_SIZE 19
const char s[STRING_SIZE] = "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff"
 (    here are   spaces    )"\xff\xff\xff";
```

Reply:
Oh right!